core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java

```diff
@@ -422,8 +422,8 @@ private ClusterCacheStatus initCacheStatusIfAbsent(String cacheName) {
          // (including the coordinator) before it starts on any node.
          AvailabilityStrategy availabilityStrategy;
          Configuration cacheConfiguration = cacheManager.getCacheConfiguration(cacheName);
-         boolean resolveConflictsOnMerge = cacheConfiguration.clustering().partitionHandling().mergePolicy() != null;
          PartitionHandling  partitionHandling = cacheConfiguration != null ? cacheConfiguration.clustering().partitionHandling().whenSplit() : null;
+         boolean resolveConflictsOnMerge = partitionHandling == PartitionHandling.ALLOW_READ_WRITES && cacheConfiguration.clustering().partitionHandling().mergePolicy() != null;
```

Comment:
If there's no default merge policy, we should throw an exception in the `ConfigurationBuilder.validate()` when the partition handling is set to ARW.